From: Galih Tama <galpt@v.recipes>
Date: Tue, 12 May 2026 22:22:09 +0700
Subject: [PATCH] Make FLOW the default I/O scheduler

Select flow-iosched as the default blk-mq I/O scheduler when
CONFIG_MQ_IOSCHED_DEFAULT_FLOW is enabled.

Must be applied after 0001-linux7.0-flow-iosched-v1.0.0.patch.

Signed-off-by: Galih Tama <galpt@v.recipes>
---
 block/elevator.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/block/elevator.c b/block/elevator.c
index d64b5cbb20..7d59c88ddc 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -632,6 +632,14 @@ static inline bool elv_support_iosched(struct request_queue *q)
 static struct elevator_type *elevator_get_default(struct request_queue *q)
 {
 	bool is_sq = q->nr_hw_queues == 1 || blk_mq_is_shared_tags(q->tag_set->flags);
+	bool try_flow = false;
+
+#ifdef CONFIG_MQ_IOSCHED_DEFAULT_FLOW
+	try_flow = true;
+#endif
+
+	if (try_flow)
+		return elevator_find_get("flow-iosched");
 
 	if (!is_sq)
 		return NULL;
